src/components/implementation/no_interface/scheddev/sl_plugins.h

```diff
@@ -48,4 +48,8 @@ void sl_mod_init(void);
 void sl_timeout_mod_expended(cycles_t now, cycles_t oldtimeout);
 void sl_timeout_mod_init(void);
 
+/* API for thread wakeup using timeout module */
```

Comment:
blocking will be done very commonly, and very uncommonly will the timeouts be used.  No, I'd rather not make the common-case API more complex (the "dependency" argument is already confusing enough).  So `sl_thd_block_timeout(...)`, though verbose, is likely better.